public/app/plugins/datasource/loki/backendResultTransformer.ts

```diff
@@ -30,6 +30,7 @@ function processStreamFrame(
     limit: query?.maxLines,
     searchWords: query !== undefined ? getHighlighterExpressionsFromQuery(formatQuery(query.expr)) : undefined,
     custom: {
+      ...frame.meta?.custom, // keep the original meta.custom
```

Comment:
i had a bug here where we just overwrote the `frame.meta.custom`  that was coming from the server.